src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,41 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private K[] arrayKey;
+    private V[] arrayValue;
+    private int size;
+
+    public StorageImpl() {
+        arrayKey = (K[]) new Object[MAX_SIZE];
+        arrayValue = (V[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (key == arrayKey[i] || arrayKey[i] != null && arrayKey[i].equals(key)) {
+                arrayValue[i] = value;
+                return;
+            }
+        }
+        arrayKey[size] = key;
+        arrayValue[size] = value;
+        size++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < MAX_SIZE; i++) {
```

Comment:
Checking up to the `size` would be enough since we know for sure that all other indexes after that will be empty
```suggestion
        for (int i = 0; i < size; i++) {
```